src/hls-adapter.js

```diff
@@ -600,6 +609,11 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
     this._responseFilterError = false;
     this._hls.detachMedia();
     this._hls.destroy();
+    if (this._loadPromiseHandlers) {
```

Comment:
momve to destroy. else `_reloadWithDirectManifest` will get rejected promise

Reply:
done